### PR TITLE
TEST/TAG: Fix tag test to check inline data flag

### PR DESCRIPTION
**Pull request by yosefe**
## Why
UCT test must copy-out inline data, following #5889 
@bureddy

### CODE REVIEW

File: test/gtest/uct/test_tag.cc [hunk 1]
```diff
@@ -388,6 +388,9 @@ class test_tag : public uct_test {
         recv_ctx *user_ctx = ucs_container_of(self, recv_ctx, uct_ctx);
         user_ctx->sw_rndv  = true;
         user_ctx->status   = status;
+        if (flags & UCT_TAG_RECV_CB_INLINE_DATA) {
```
Comment (dmitrygx) [inline test/gtest/uct/test_tag.cc:391]: [minor/imp] can reduce duplication:
```
static void sw_rndv_completed(uct_tag_context_t *self, uct_tag_t stag,
                              const void *header, unsigned header_length,
                              ucs_status_t status, unsigned flags)
{
    recv_ctx *user_ctx = ucs_container_of(self, recv_ctx, uct_ctx);
    user_ctx->sw_rndv  = true;
    if (flags & UCT_TAG_RECV_CB_INLINE_DATA) {
        memcpy(user_ctx->mbuf->ptr(), header, header_length);
    }
    completed(self, stag, 0, header_length, NULL, status);
}
```

**Review by dmitrygx — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Comment by dmitrygx** [comment]
bot:pipe:retest

**Review by brminich — APPROVED** [review]

**Review by bureddy — APPROVED** [review]

**Comment by yosefe** [comment]
merging since test failure is unrelated

**Pull request merged**